### PR TITLE
Add dummy implementation of syscall sched_getaffinity

### PR DESCRIPTION
**Pull request by StanPlatinum**
This would help #703 .

### CODE REVIEW

File: kernel/aster-nix/src/syscall/sched_getaffinity.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            match process_table::get_process(&pid) {
+                Some(_process) => { /* Placeholder if process-specific logic needed */ }
+                None => return Err(Error::with_message(Errno::ESRCH, "process does not exist")),
+            }
+        }
+    }
+
+    // Assuming all CPUs are available; construct the cpu_set_t accordingly.
+    let result = cpu_set_t {
+        __bits: [usize::MAX; CPU_SETSIZE / __NCPUBITS],
```
Comment (tatetian) [inline kernel/aster-nix/src/syscall/sched_getaffinity.rs:39]: I think the number of bit 1 should be equal to the number of CPUs when all CPUs are available to schedule.

File: kernel/aster-nix/src/syscall/sched_getaffinity.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    process::{process_table, Pid},
+    util::write_val_to_user,
+};
+
+pub fn sys_sched_getaffinity(
+    pid: Pid,
+    cpuset_size: usize,
+    user_mask_ptr: Vaddr,
+) -> Result<SyscallReturn> {
+    // Ensure cpuset_size is sufficient to store a cpu_set_t.
```
Comment (tatetian) [inline kernel/aster-nix/src/syscall/sched_getaffinity.rs:17]: Please remove worthless comments. They do not provide any extra information or provide information in a more comprehensible way than the code.

```rust
// Ensure cpuset_size is sufficient to store a cpu_set_t.

// Use match statement for clearer control flow especially when pid == 0 logic is implemented.

// Check if the process exists, in a Rust idiomatic way using match for clarity.

// Assuming all CPUs are available; construct the cpu_set_t accordingly.
```

File: kernel/aster-nix/src/syscall/sched_getaffinity.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            // Check if the process exists, in a Rust idiomatic way using match for clarity.
+            match process_table::get_process(&pid) {
+                Some(_process) => { /* Placeholder if process-specific logic needed */ }
+                None => return Err(Error::with_message(Errno::ESRCH, "process does not exist")),
+            }
+        }
+    }
+
+    // Assuming all CPUs are available; construct the cpu_set_t accordingly.
+    let result = cpu_set_t {
```
Comment (tatetian) [inline kernel/aster-nix/src/syscall/sched_getaffinity.rs:38]: Specific names are better than general ones like `result`.

File: kernel/aster-nix/src/syscall/sched_getaffinity.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+use crate::{
+    prelude::*,
+    process::{process_table, Pid},
+    util::write_val_to_user,
+};
+
+pub fn sys_sched_getaffinity(
+    pid: Pid,
+    cpuset_size: usize,
+    user_mask_ptr: Vaddr,
```
Comment (tatetian) [inline kernel/aster-nix/src/syscall/sched_getaffinity.rs:15]: Rename this variable to `cpu_set_ptr` to be consistent with the type `cpu_set_t`. Rename the `result` variable to `cpu_set` or `dummy_cpu_set` for the same reason.

**Review by tatetian — COMMENTED** [review]
Remember to remove the binary `sched_getaffinity`.

File: kernel/aster-nix/src/syscall/sched_getaffinity.rs (hunk 2 above)
Comment (StanPlatinum) [reply kernel/aster-nix/src/syscall/sched_getaffinity.rs:17]: Done.

File: kernel/aster-nix/src/syscall/sched_getaffinity.rs (hunk 3 above)
Comment (StanPlatinum) [reply kernel/aster-nix/src/syscall/sched_getaffinity.rs:38]: I have renamed it as `cpu_affinity_set`.

File: kernel/aster-nix/src/syscall/sched_getaffinity.rs (hunk 4 above)
Comment (StanPlatinum) [reply kernel/aster-nix/src/syscall/sched_getaffinity.rs:15]: Done.

File: kernel/aster-nix/src/syscall/sched_getaffinity.rs (hunk 1 above)
Comment (StanPlatinum) [reply kernel/aster-nix/src/syscall/sched_getaffinity.rs:39]: Thanks for the advice! I have revised this part.

The function `fn get_num_cpus() -> usize` provides a mechanism to determine the actual number of CPUs available on a system at runtime.

The `cpu_set_t::new` function creates a precise CPU affinity mask based on num_cpus, which signifies the desired number of available CPUs for scheduling.
This implementation iterates over CPU indices up to `num_cpus` or `CPU_SETSIZE`. For each index, it sets a corresponding bit to 1 in the `__bits` array, reflecting the actual CPU availability.

**Comment by StanPlatinum** [comment]
> Remember to remove the binary `sched_getaffinity`.

Done. Thanks!

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**